src/components/mongo.ts

```diff
@@ -0,0 +1,409 @@
+            argsWithDefaults.port,
+            argsWithDefaults.environment,
+            argsWithDefaults.secrets,
+            this.logGroup.name,
+            awsRegion,
+          ])
+          .apply(
+            ([
+              containerName,
+              image,
+              port,
+              environment,
+              secrets,
+              logGroup,
+              region,
+            ]) => {
+              return JSON.stringify([
+                {
+                  readonlyRootFilesystem: false,
+                  name: containerName,
+                  image,
+                  essential: true,
+                  portMappings: [
+                    {
+                      containerPort: port,
+                      protocol: 'tcp',
+                    },
+                  ],
+                  mountPoints: [
+                    {
+                      containerPath: '/data/db',
+                      sourceVolume: `${this.name}-volume`,
+                    },
+                  ],
+                  logConfiguration: {
+                    logDriver: 'awslogs',
+                    options: {
+                      'awslogs-group': logGroup,
+                      'awslogs-region': region,
+                      'awslogs-stream-prefix': 'ecs',
+                    },
+                  },
+                  environment,
+                  secrets,
+                },
+              ] as ContainerDefinition[]);
+            },
+          ),
+        volumes: [
+          {
+            name: `${this.name}-volume`,
+            efsVolumeConfiguration: {
+              fileSystemId: this.mountTarget.fileSystemId,
+              transitEncryption: 'ENABLED',
+            },
+          },
+        ],
+        tags: { ...commonTags, ...argsWithDefaults.tags },
+      },
+      { parent: this },
+    );
+
+    return taskDefinition;
+  }
+
+  private createServiceDiscovery(args: MongoArgs) {
+    const privateDnsNamespace = new aws.servicediscovery.PrivateDnsNamespace(
+      `${this.name}-private-dns-namespace`,
+      {
+        vpc: args.vpc.vpcId,
+      },
+    );
+
+    return new aws.servicediscovery.Service(`${this.name}-service-discovery`, {
+      dnsConfig: {
+        namespaceId: privateDnsNamespace.id,
+        dnsRecords: [
+          {
+            ttl: 10,
+            type: 'A',
+          },
+        ],
+        routingPolicy: 'MULTIVALUE',
+      },
+      healthCheckCustomConfig: {
+        failureThreshold: 1,
+      },
+    });
+  }
+
+  private createEcsService(args: MongoArgs) {
+    const argsWithDefaults = Object.assign({}, defaults, args);
+
+    const service = new aws.ecs.Service(
+      `${this.name}-service`,
+      {
+        name: this.name,
+        cluster: argsWithDefaults.cluster.id,
+        launchType: 'FARGATE',
+        desiredCount: argsWithDefaults.desiredCount,
+        taskDefinition: this.taskDefinition.arn,
+        enableExecuteCommand: true,
```

Comment:
We can probably disable the option to exec into mongo container, can you think of any potential use case where this will be needed?